### DISCUSSION TITLE
How to pass parameters to Github Actions Workflow

### DISCUSSION THREAD

**Discussion by missedone** [community/community#27002] [Actions]
<p>hi folks</p>
<p>I have bunch of <a href="https://wiki.jenkins.io/display/JENKINS/Parameterized+Build" rel="nofollow noopener">parameterized Jenkins jobs</a>, saying before starting the build, the user can input the params for the job.</p>
<p>i’m looking for similar feature for Github Actions for our deployment job which listen on github deployment event.</p>
<p>for example, i’d like to allow user to be able to turn on/off some flags of the deployment, </p>
<p>looks like the github deployment api support payload with any value like a json string: <a href="https://developer.github.com/v3/repos/deployments/#create-a-deployment" rel="nofollow noopener">https://developer.github.com/v3/repos/deployments/#create-a-deployment</a></p>
<p>can the Github Actions receive that deployment payload and extract values for the workflow?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/missedone">@missedone</a> , </p>
<p>I am not sure whether I fully understood your requirement. Based on my understanding , you want github workflow to receive deployment evetnt payload and use the payload content in your workflow run. </p>
<p>There is a <a href="https://help.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#github-context" rel="nofollow noopener"><strong>github</strong> context</a> for you to use which includes the deployment event payload.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/d91a8d28e7783e88320f62d78eb758337524e2e8.png" data-download-href="/uploads/short-url/uYAuPhl4Nj1YiG9sQ16ZVVY5kdW.png?dl=1" title="deployment event.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181101123-d31e84bc-1623-44f1-ab08-529b6d4c5bdd.png" alt="deployment event.png" data-base62-sha1="uYAuPhl4Nj1YiG9sQ16ZVVY5kdW" width="690" height="429" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/d91a8d28e7783e88320f62d78eb758337524e2e8_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">deployment event.png</span><span class="informations">1088×677 23.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>And you could use ${{ }} syntax to get the value of json key value. Such as </p>
<p><strong>${{ github.event.deployment.environment }} </strong></p>
<p>There is an example of workflow yml to get environment name: </p>
<pre><code>name: deployment
on:
  deployment
jobs:
  build:
    runs-on: ubuntu-latest
    steps:
     - name: Dump GitHub context
       env:
          GITHUB_CONTEXT: ${{ tojson(github) }}
          ENVIRONMENT: ${{ github.event.deployment.environment }}
       run: |
          echo "$GITHUB_CONTEXT"
          echo $ENVIRONMENT</code></pre>

**Comment by missedone** [comment]
<p>right, i wasn’t aware of this, the github context works for me,</p>
<p>the payload can be any format, so i use jq in shell to extract the value:</p>
<pre><code class="lang-auto">
DEPLOY\_PAYLOAD\_JSON='${{ github.event.deployment.payload }}'

echo $DEPLOY\_PAYLOAD\_JSON | jq -r \&lt;the\_json\_path\&gt;

</code></pre>
<p>thanks.</p>

**Comment by joeflack4** [comment]
There are ways to do this using [workflow_dispatch](https://docs.github.com/en/enterprise-cloud@latest/actions/using-workflows/workflow-syntax-for-github-actions), if that works for your use case. I tried with repository_dispatch also myself, but it doesn't seem like it accepts inputs, or if so, it's not (well) documented.

Check the link above to see how to format your `.yaml`. There are lots of examples you can find online of `curl` for this calling such an action. Here's how I did mine in Python:

```py
    headers = {
        "Authorization": f"Bearer {CONFIG['personal_access_token']}",
        "Accept": "application/vnd.github.v3+json",
        "Content-Type": "application/json"
    }
    ref = 'develop'
    workflow_id = 'resolve_fetch_failures_0_members.yml'
    url = f'https://api.github.com/repos/jhu-bids/TermHub/actions/workflows/{workflow_id}/dispatches'
    payload = {'ref': ref}
    if params:
        payload['inputs'] = params
    response = requests.post(url, headers = headers,data=json.dumps(payload))
```
